code/atom.dm

```diff
@@ -241,21 +242,25 @@ TYPEINFO(/atom)
 	proc/return_air()
 		return null
 	/**
-	  * Convenience proc to see if a container is open for chemistry handling
-	  * Takes an argument of whether this openness is for the purpose of pouring something in or not (this should maybe just be a separate flag but we ran out of bits okay)
-	  * returns true if open, false if closed
+	  * Convenience proc to see if a container is open for chemistry handling.
+	  * - Takes arguments of whether this openness is for the purpose of pouring something in or out.
+	  * - `TRUE` means you're checking chems go in. `FALSE` means pouring outward. Leaving it null means checking either way.
+	  * - returns `TRUE` if you can access insides, `FALSE` if closed.
+	  * - child procs may override this for special behaviours, such as checking size.
 	  */
-	proc/is_open_container(input = FALSE)
-		return flags & OPENCONTAINER
-
-	/// Set a container to be open or closed and handle chemistry reactions that might happen as a result
-	proc/set_open_container(value)
-		if (value)
-			ADD_FLAG(src.flags, OPENCONTAINER)
+	proc/is_open_container(inward)
+		CRASH("You're meant to override is_open_container()! The culprit: [src]")
```

Comment:
Ok I don't quite get why this CRASH-es by default. I would expect the default behaviour to use `chem_flags`? What's the point of having `chem_flags` if they do not actually affect this proc? Is it overriden to use `chem_flags` somewhere? If so I don't see where but also there are 130 files to go through so

Reply:
holdover from a previous comment from @TobleroneSwordfish 
![image](https://github.com/goonstation/goonstation/assets/51260013/5393813d-4ea2-496b-8683-c911b51c13df)
I could make the default case work using those flags now.

Reply:
I've made the default case check flags now.